### ISSUE TITLE
Doubt about ARRAY() in AlaSQL

### ISSUE THREAD

**Issue by podgorski** [agershun/alasql#400]
Hi,

Someone could clear my doubt about ARRAY() in AlaSQL?

I would like to know how to group a plain object  in an array nested object with more than one item.

My Input:

var origArr = [
    {food: 'apple', type: 'fruit', color:'red'},
    {food: 'potato', type: 'vegetable', color:'yellow'},
    {food: 'banana', type: 'fruit', color: 'yellow'}
];

the desired output:

[
    { type: 'fruit', foods: [ {food: 'apple',  color:'red'}, {food: 'banana,  color: 'yellow'} ] },
    { type: 'vegetable', foods: [ {food: 'potato',  color:'yellow'} ] },    
];

when array needs only one item thats ok:

    alasql('SELECT type, ARRAY(food) AS foods FROM ? GROUP BY type', [origArr]);

However, if the array needs be a object of properties I don't figure out how to do.

Thank you.

Best Regards.

**Comment by agershun** [comment]
Please use ```_``` pseudofield like below:
```js
    var res2 = alasql('SELECT type, ARRAY(_) AS foods FROM ? GROUP BY type', [origArr]);
```
You can try this example [in jsFiddle](http://jsfiddle.net/agershun/1nt5afd3/1/)

**Comment by gnachao** [comment]
Very interesting!

I was playing around but couldn't figure it out. As playing, I was hoping the following code working:

```javascript
 var res2 = alasql('SELECT type, ARRAY({food, color}) AS foods FROM ? GROUP BY type', [origArr]);
```

**Comment by agershun** [comment]
Only one things: not ```{food, color}```, but ```{food:food, color:color}```

```js
var res2 = alasql('SELECT type, ARRAY({food:food, color:color}) AS foods FROM ? GROUP BY type', [origArr]);
```

Please, see the example [here](http://jsfiddle.net/agershun/1nt5afd3/2/)

**Comment by podgorski** [comment]
Thank you agershun for quickly reply,  the second answer works like a charm.

var res2 = alasql('SELECT type, ARRAY({food:food, color:color}) AS foods FROM ? GROUP BY type', [origArr]);

Congratulations, alasql is so elegant solution.

**Comment by gnachao** [comment]
Wow! Thank you!

**Comment by podgorski** [comment]
agersun another question:

If outside the array I desired show more than property that I Group by:

for example:

var origArr = [
{food: 'apple', type: 'fruit', color:'red', type_id:1,... (more items)},
{food: 'potato', type: 'vegetable', color:'yellow', type_id:2,... (more items)},
{food: 'banana', type: 'fruit', color: 'yellow', type_id:1,... (more items)}
];

the desired answer:

[
{"type":"fruit","type_id":1,more items,"foods":[{"food":"apple","color":"red"},{"food":"banana","color":"yellow"}] },
{"type":"vegetable","type_id":2,more items,"foods":[{"food":"potato","color":"yellow"}]}
]

var res2 = alasql('SELECT type, ARRAY({food:food, color:color}) AS foods FROM ? GROUP BY type', [origArr]);

I know I could Group By, however if I have other properties.

Could you clarify this point for me?

**Comment by podgorski** [comment]
I figure out here

var alasql = require('alasql');

var origArr = [
    {id:10, nome:'Antonio',servico:'Arranca Dedo',serv_id:'21',produto:'Homeoclean',prod_id:'31'},
    {id:10, nome:'Antonio',servico:'Arranca Pé',serv_id:'22',produto:'Homeoclean',prod_id:'31'},
    {id:11, nome:'Pedro',servico:'Arranca Dedo',serv_id:'21',produto:'Homeoclean',prod_id:'31'},
    {id:11, nome:'Pedro',servico:'Arranca Dedo',serv_id:'21',produto:'Talco xulezinho',prod_id:'32'}
    
];

var fields = "id,nome";

var res3 = alasql('SELECT ' + fields + ', ARRAY({serv_id:serv_id, servico:servico}) AS servicos, ARRAY({prod_id:prod_id, produto:produto}) AS produtos FROM ? GROUP BY ' + fields, [origArr]);

console.log(res3);

----
Thank you for your time.

**Comment by agershun** [comment]
You can use FIRST() aggregator, like:
```sql
    SELECT id, FIRST(nome), ARRAY(...) FROM ... GROUP BY id
```
Here you can group by ```id``` field, but for the name (nome) you use first occurrence of this field.

BTW You can use DISTINCT for aggregator:
```sql
SELECT ARRAY(DISTINCT {prod_id:prod_id}) FROM ...
```
to collect only distinct values for services and products.

Is this answer on your question?

**Comment by podgorski** [comment]
Superb agershun, 

exactly what I need.

One more time, thank you for your time.

**Comment by mathiasrw** [comment]
I will place this in the documentation

**Comment by agershun** [comment]
@mathiasrw  :+1:

**Comment by mathiasrw** [comment]
Closed per request from #628
